modules/seqexec/server/src/main/scala/seqexec/server/SequenceGen.scala

```diff
@@ -3,23 +3,54 @@
 
 package seqexec.server
 
+import cats.effect.IO
 import cats.implicits._
 import gem.Observation
 import seqexec.model.StepConfig
 import seqexec.model.enum.{Instrument, Resource}
 import seqexec.engine.{Step => EngineStep}
+import seqexec.model.dhs.ImageFileId
 
 /*
  * SequenceGen keeps all the information extracted from the ODB sequence.
- * It is combined with header parameters to build an engine.Sequence. It allows to rebuild the engine
- * whenever any of those parameters change.
+ * It is combined with header parameters to build an engine.Sequence. It allows to rebuild the
+ * engine sequence whenever any of those parameters change.
  */
 final case class SequenceGen(id: Observation.Id, title: String, instrument: Instrument, steps: List[SequenceGen.Step]) {
-  val resources: Set[Resource] = steps.foldMap(_.resources)
+  val resources: Set[Resource] = steps.collect{
+    case SequenceGen.PendingStep(_, _, resources, _) => resources
+  }.foldMap(identity(_))
 }
 
 object SequenceGen {
 
-  final case class Step(id: Int, config: StepConfig, resources: Set[Resource], generator: HeaderExtraData => EngineStep)
+  sealed trait Step {
```

Comment:
We have a bunch of `Step` classes defined, could we have something else `StepGen` or something?